Sources/Utility/ArrayExtensions.swift

```diff
@@ -41,4 +41,8 @@ extension Array {
         }
         return (a, b)
     }
+    
+    public func nilIfEmpty() -> Array? {
+        return self.isEmpty ? nil : self
+    }
```

Comment:
nit-pick: You only use this method in one place. I'm worried other developers won't know this exists and end up duplicating this logic elsewhere. Why not inline it for now, and extract it into its own method if the need ever arises later?